hoprd/db/api/src/metadata.rs

```diff
@@ -0,0 +1,263 @@
+use async_trait::async_trait;
+use sea_orm::{ColumnTrait, EntityTrait, QueryFilter};
+use serde::{Deserialize, Serialize};
+
+use crate::{db::HoprdDb, errors::Result};
+
+/// Represents an alias
+#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize)]
+pub struct AliasEntry {
+    pub peer_id: String,
+    pub alias: String,
+}
+
+impl AliasEntry {
+    pub fn new(peer_id: String, alias: String) -> Self {
+        Self { peer_id, alias }
+    }
+}
+
+/// Defines DB API for accessing HOPR settings (mainly aliases for now)
+#[async_trait]
+pub trait HoprdDbMetadataOperations {
+    /// Retrieve the alias for a given PeerID
+    async fn resolve_alias(&self, alias: String) -> Result<Option<String>>;
```

Comment:
```suggestion
    async fn resolve(&self, alias: String) -> Result<Option<String>>;
```

Reply:
Should be distinguished by the function name, because once you bring the trait into scope, it won't be apparent what `resolve` function you are calling. 

So I'd dismiss this suggestion.